modules/core/arrow-test/src/main/kotlin/arrow/test/laws/TimerLaws.kt

```diff
@@ -0,0 +1,56 @@
+package arrow.test.laws
+
+import arrow.Kind
+import arrow.effects.Timer
+import arrow.effects.typeclasses.Async
+import arrow.effects.typeclasses.milliseconds
+import arrow.effects.typeclasses.seconds
+import arrow.typeclasses.Eq
+
+object TimerLaws {
+
+  // TODO move to Arrow-effects and figure out a acceptable API for timeMilis/timeNano for MPP.
+  interface Clock<F> {
+    fun timeMillis(): Kind<F, Long>
+    fun timeNano(): Kind<F, Long>
+
+    companion object {
+      operator fun <F> invoke(AS: Async<F>): Clock<F> = object : Clock<F> {
+        override fun timeMillis(): Kind<F, Long> =
+          AS.effect { System.currentTimeMillis() }
+        override fun timeNano(): Kind<F, Long> =
+          AS.effect { System.nanoTime() }
+      }
+    }
+  }
+
+  fun <F> laws(AS: Async<F>, T: Timer<F>, EQ: Eq<Kind<F, Boolean>>): List<Law> =
+    listOf(
+      Law("Timer Laws: sleep should last specified time") { AS.sleepShouldLastSpecifiedTime(T, Clock(AS), EQ) },
+      Law("Timer Laws: negative sleep should be immediate") { AS.negativeSleepShouldBeImmediate(T, EQ) }
+    )
+
+  fun <F> Async<F>.sleepShouldLastSpecifiedTime(
+    T: Timer<F>,
+    C: Clock<F>,
+    EQ: Eq<Kind<F, Boolean>>
+  ) {
+    val lhs = binding {
```

Comment:
I'd test it multiple times to prevent issues. `forFew(10` or something like that.